.github/workflows/cd.yml

```diff
@@ -206,33 +206,38 @@ jobs:
     steps:
       - name: Check environment
         run: |
-          if [ "${{ inputs.environment }}" != 'dev' ] && [ "${{ inputs.branch }}" != 'main' ]; then
+          if [ "${ENVIRONMENT}" != 'dev' ] && [ "${BRANCH}" != 'main' ]; then
             echo "Only 'dev' environment is allowed for non-main branches."
             exit 1
           fi
 
-          if [ "${{ inputs.docs-only }}" == 'true' ]; then
-            if [ "${{ inputs.environment }}" != 'prod' ]; then
+          if [ "${DOCS_ONLY}" == 'true' ]; then
+            if [ "${ENVIRONMENT}" != 'prod' ]; then
               echo "Only 'prod' environment is allowed for docs publishing."
               exit 1
             fi
 
-            if [ "${{ inputs.branch }}" != 'main' ]; then
+            if [ "${BRANCH}" != 'main' ]; then
               echo "Only 'main' branch can be used for docs publishing."
               exit 1
             fi
           fi
+        env:
+          ENVIRONMENT: ${{ inputs.environment }}
+          BRANCH: ${{ inputs.branch }}
+          DOCS_ONLY: ${{ inputs.docs-only }}
         shell: bash
 
       - name: Checkout specified branch
         id: checkout-specified-branch
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           ref: ${{ inputs.branch }}
+          persist-credentials: false
 
   ci:
     name: CI
-    uses: grafana/plugin-ci-workflows/.github/workflows/ci.yml@main
+    uses: grafana/plugin-ci-workflows/.github/workflows/ci.yml@giuseppe/gha-zizmor # zizmor: ignore[unpinned-uses]
```

Comment:
is this branch something final or are you planing to change it back to `main` once you have tested it?

Reply:
Oops, testing only. I'll revert